tests/cephadm/test_client.py

```diff
@@ -0,0 +1,149 @@
+"""Manage the client via cephadm CLI."""
+import logging
+from typing import Dict
+
+from ceph.ceph_admin.orch import Orch
+from ceph.utils import get_node_by_id
+
+log = logging.getLogger(__name__)
+
+
+def add(cls, config: Dict) -> None:
+    """
+    configure client using the provided configuration.
+
+    Args:
+        cls: cephadm object
+        config: Key/value pairs provided by the test case to create the client.
+
+    Example
+        config:
+            command: add
+            id: client.1                    # client Id
+            node: "node8"                   # client node
+            install_packages:
+              - ceph_common                 # install ceph common packages
+            copy_admin_keyring: true|false  # copy admin keyring
+            caps:                           # authorize client capabilities
+              - "mon 'allow r'"
+              - "osd 'allow rw pool=liverpool'"
+    """
+    id_ = config["id"]
+    node = get_node_by_id(cls.cluster, config["node"])
+
+    def put_file(client, file_name, content, file_mode, sudo=True):
+        file_ = client.remote_file(sudo=sudo, file_name=file_name, file_mode=file_mode)
+        file_.write(content)
+        file_.flush()
+
+    # Create client
+    cmd = ["ceph", "auth", "get-or-create", f"{id_}"]
+    [cmd.append(f"{k} '{v}'") for k, v in config.get("caps").items()]
+    cnt_key, err = cls.shell(args=cmd)
+
+    # Copy the keyring to client
+    node.exec_command(sudo=True, cmd="mkdir -p /etc/ceph")
+    put_file(node, f"/etc/ceph/ceph.{id_}.keyring", cnt_key, "w")
+
+    # Get minimal ceph.conf
+    ceph_conf, err = cls.shell(args=["ceph", "config", "generate-minimal-conf"])
+
+    # Copy the ceph.conf to client
+    put_file(node, "/etc/ceph/ceph.conf", ceph_conf, "w")
+
+    # Copy admin keyring to client node
```

Comment:
Shouldn't we check if `copy_admin_keyring` is requested or not?

Reply:
Its needed. updated now.